bundles/org.eclipse.equinox.common/src/org/eclipse/core/runtime/URIUtil.java

```diff
@@ -328,6 +330,33 @@ public static URI toURI(URL url) throws URISyntaxException {
 		}
 	}
 
+	/**
+	 * Returns the {@code Path} specified by the given {@code file} {@link URL}.
+	 * 
+	 * @throws IllegalArgumentException if the specified URL does not use the
+	 *                                  {@code file} protocol
+	 * 
+	 * @since 3.20
+	 */
+	public static Path toFilePath(URL url) {
```

Comment:
That the name contains `File` and ` Path` seems to be a duplication at first, but since any kind of URL can have a path I think it should be made clear that it's a file-system path (although nio-Path can also handle non-classical file-systems).
On the other hand `toFileSystemPath()` seemed to be a bit verbose to me.